src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -184,6 +184,20 @@ private static MarshallerKind GetMarshallerKind(TypeDesc type, MarshalAsDescript
             {
                 return MarshallerKind.SafeHandle;
             }
+
+            // Temporary fix for out SafeHandle scenario
+            // TODO: handle in,out,ref properly
+            if (paramMetadata.Out && !paramMetadata.In)
+            {
+                Debug.Assert(type is ByRefType);
```

Comment:
I understand that this is temporary ... but this should throw instead. I do not see anything that would guarantee this to be ByRefType.